Appropriation Art Exhibit/exhibit.py

```diff
@@ -0,0 +1,697 @@
+__author__ = 'Harriet'
+
+import random
+import math
+import os
+
+from PIL import Image
+
+
+# The number of colour components, in this case, 3 (red, green and blue)
+NUMBER_OF_COLOR_COMPONENTS = 3
+
+RED = (255,0,0)
+GREEN = (0, 255, 0)
+BLUE = (0, 0, 255)
+BLACK = (0, 0, 0)
+
+
+def get_distance(first_point, second_point):
+    """Return distance between 2 coordinates as a float
+
+    Arguments:
+    first_point -- first coordinate tuple
+    second_point -- second coordinate tuple
+    """
+    x_distance = second_point[0] - first_point[0]
+    y_distance = second_point[1] - first_point[1]
+    distance = math.sqrt(x_distance**2.0 + y_distance**2.0)
+    return distance
+
+
+def show_gallery():
+    input_dir = "source-images"
+    output_dir = "output-images"
+
+    filenames = ["alf.png",
+                 "hug.png",
+                 "sad.jpg",
+                 "jegermeister.jpg"]
+
+    paintings = []
+    for filename in filenames:
+        paintings.append(Painting(os.path.join(input_dir, filename)))
+
+    effects = []
+    colors = (Color(150, 0, 150),
+              Color(150, 150, 0),
+              Color(0, 150, 0),
+              Color(0, 150, 150))
+    tile_effect = TileEffect(colors, 6, 2)
+    effects.append(tile_effect)
+
+    dot_effect = DotEffect(10, 5, Color(*BLACK))
+    effects.append(dot_effect)
+
+    shuffle_effect = ShuffleEffect(10, 3)
+    effects.append(shuffle_effect)
+
+    colors = (Color(255, 0, 255),
+              Color(255, 255, 0),
+              Color(0, 255, 255))
+    colorchange_effect = ThreeColorEffect(50, 0.9, colors)
+    effects.append(colorchange_effect)
+
+    for i in range(len(effects)):
+        effects[i].do_effect(paintings[i])
+        paintings[i].show()
+
+    for i in range(len(paintings)):
+        paintings[i].save(os.path.join(output_dir, filenames[i]))
+
+
+class Point(object):
```

Comment:
The structure of the code is very tidy and clear when reading through it. If editing were to be required then after understanding what each section does I can see that it should be a simple task to modify and maintain.